### PR TITLE
feat: deploy utils cleanup

### PR DESCRIPTION
**Pull request by smartcontracts**
Updates DeployUtils to include new utils for using create and
create2.

### CODE REVIEW

File: packages/contracts-bedrock/scripts/deploy/Deploy.s.sol [hunk 1]
```diff
@@ -665,7 +667,13 @@ contract Deploy is Deployer {
 
     /// @notice Deploy the SuperchainConfig contract
     function deploySuperchainConfig() public broadcast {
-        ISuperchainConfig superchainConfig = ISuperchainConfig(_deploy("SuperchainConfig", hex""));
+        ISuperchainConfig superchainConfig = ISuperchainConfig(
+            this.create2AndSave({
```
Comment (smartcontracts) [inline packages/contracts-bedrock/scripts/deploy/Deploy.s.sol]: this is a bit too deranged for my taste, I'll probably replace it with:

```
DeployUtils.create2AndSave({
  _artifacts: this,
  _name: "SuperchainConfig",
  _args: abi.encodeCall(ISuperchainConfig.__constructor__, ()),
  _salt: _implSalt()
})
```

File: packages/contracts-bedrock/scripts/libraries/DeployUtils.sol [hunk 2]
```diff
@@ -1,27 +1,212 @@
...
 
 library DeployUtils {
     Vm internal constant vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
 
-    // This takes a sender and an identifier and returns a deterministic address based on the two.
-    // The result is used to etch the input and output contracts to a deterministic address based on
-    // those two values, where the identifier represents the input or output contract, such as
-    // `optimism.DeploySuperchainInput` or `optimism.DeployOPChainOutput`.
-    // Example: `toIOAddress(msg.sender, "optimism.DeploySuperchainInput")`
+    /// @notice Deploys a contract the given name and arguments via CREATE.
```
Comment (AmadiMichael) [inline packages/contracts-bedrock/scripts/libraries/DeployUtils.sol]:
```suggestion
    /// @notice Deploys a contract with the given name and arguments via CREATE.
```

File: packages/contracts-bedrock/scripts/libraries/DeployUtils.sol [hunk 3]
```diff
@@ -1,27 +1,212 @@
...
+    }
+
+    /// @notice Deploys a contract with the given name via CREATE.
+    /// @param _name Name of the contract to deploy.
+    /// @return Address of the deployed contract.
+    function create1(string memory _name) internal returns (address payable) {
+        return create1(_name, hex"");
+    }
+
+    /// @notice Deploys a contract with the given name via CREATE and saves the result.
```
Comment (AmadiMichael) [inline packages/contracts-bedrock/scripts/libraries/DeployUtils.sol]:
```suggestion
    /// @notice Deploys a contract with the given name and arguments via CREATE and saves the result.
```

File: packages/contracts-bedrock/scripts/libraries/DeployUtils.sol [hunk 4]
```diff
@@ -1,27 +1,212 @@
...
+        string memory _name,
+        string memory _nickname
+    )
+        internal
+        returns (address payable addr_)
+    {
+        return create1AndSave(_save, _name, _nickname, hex"");
+    }
+
+    /// @notice Deploys a contract with the given name via CREATE and saves the result.
```
Comment (AmadiMichael) [inline packages/contracts-bedrock/scripts/libraries/DeployUtils.sol]:
```suggestion
    /// @notice Deploys a contract with the given name and arguments via CREATE and saves the result.
```

**Review by AmadiMichael — APPROVED** [review]

**Review by Abuchtela — COMMENTED** [review]
packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

**Pull request merged**